File: src/main/java/github/zmilla93/gui/setup/StashFolderSetupPanel.java
package github.zmilla93.gui.setup;

import github.zmilla93.core.managers.SaveManager;
import github.zmilla93.core.poe.Game;
import github.zmilla93.gui.components.poe.ThemeLabel;
import github.zmilla93.modules.theme.OLD_ThemeColor;

import javax.swing.*;

public class StashFolderSetupPanel extends AbstractSetupPanel {

    private final JRadioButton poe1YesRadioButton = new JRadioButton("Yes");
    private final JRadioButton poe2YesRadioButton = new JRadioButton("Yes");
    private final JRadioButton poe1NoRadioButton = new JRadioButton("No");
    private final JRadioButton poe2NoRadioButton = new JRadioButton("No");

    public StashFolderSetupPanel() {
        ButtonGroup poe1ButtonGroup = new ButtonGroup();
        poe1ButtonGroup.add(poe1YesRadioButton);
        poe1ButtonGroup.add(poe1NoRadioButton);

        ButtonGroup poe2ButtonGroup = new ButtonGroup();
        poe2ButtonGroup.add(poe2YesRadioButton);
        poe2ButtonGroup.add(poe2NoRadioButton);

        addHeader("Stash Tab Folders");
        addComponent(new JLabel("Do you use any stash tab folders? Doing so affects UI alignment."));
        addComponent(new ThemeLabel(OLD_ThemeColor.INDETERMINATE, "This settings needs to be updated manually if changed."));
        addVerticalStrut();
        addHeader(Game.PATH_OF_EXILE_1.explicitName);
        addComponent(poe1YesRadioButton);
        addComponent(poe1NoRadioButton);
        addVerticalStrut();
        addHeader(Game.PATH_OF_EXILE_2.toString());
        addComponent(poe2YesRadioButton);
        addComponent(poe2NoRadioButton);
    }

    @Override
    protected void addComponentListeners() {
        poe1YesRadioButton.addActionListener(e -> runSetupValidation());
        poe2YesRadioButton.addActionListener(e -> runSetupValidation());
        poe1NoRadioButton.addActionListener(e -> runSetupValidation());
        poe2NoRadioButton.addActionListener(e -> runSetupValidation());
    }

    @Override
    public void initializeComponents() {
        if (SaveManager.settingsSaveFile.data.hasInitUsingStashFolders) {
            if (SaveManager.settingsSaveFile.data.settingsPoe1.usingStashFolder) poe1YesRadioButton.setSelected(true);
            else poe1NoRadioButton.setSelected(true);
            if (SaveManager.settingsSaveFile.data.settingsPoe2.usingStashFolder) poe2YesRadioButton.setSelected(true);
            else poe2NoRadioButton.setSelected(true);
        }
    }

    @Override
    public boolean isSetupValid() {
        boolean poe1Valid = poe1YesRadioButton.isSelected() || poe1NoRadioButton.isSelected();
        boolean poe2Valid = poe2YesRadioButton.isSelected() || poe2NoRadioButton.isSelected();
        return poe1Valid && poe2Valid;
    }

    @Override
    public void applyCompletedSetup() {
        SaveManager.settingsSaveFile.data.settingsPoe1.usingStashFolder = poe1YesRadioButton.isSelected();
        SaveManager.settingsSaveFile.data.settingsPoe2.usingStashFolder = poe2YesRadioButton.isSelected();
        SaveManager.settingsSaveFile.data.hasInitUsingStashFolders = true;
    }

}

File: src/main/java/github/zmilla93/gui/setup/StashFolderSetupPanel.kt
package github.zmilla93.gui.setup

import github.zmilla93.core.managers.SaveManager
import github.zmilla93.core.poe.Game
import github.zmilla93.gui.managers.FrameManager
import github.zmilla93.modules.zswing.extensions.ActionExtensions.onToggle
import github.zmilla93.modules.zswing.theme.ThemeColorBlind
import io.github.zmilla93.modules.theme.UIProperty.FontColorExtensions.textColor
import javax.swing.JCheckBox
import javax.swing.JLabel

class StashFolderSetupPanel : AbstractSetupPanel() {

    private val poe1UsingFoldersCheckbox = JCheckBox("Using stash subfolders")
    private val poe1tradeTabsInsideFolder = JCheckBox("Trade tabs are in subfolders")

    private val poe2UsingFoldersCheckbox = JCheckBox("Using any subfolders")
    private val poe2tradeTabsInsideFolder = JCheckBox("Trade tabs are in subfolders")

    init {
        poe1tradeTabsInsideFolder.isEnabled = false
        poe2tradeTabsInsideFolder.isEnabled = false

        addHeader("Stash Tab Folders")
        addComponent(JLabel("Do you use stash subfolders? Doing so affects UI alignment."))
        addComponent(JLabel("This settings needs to be updated manually if changed.").textColor(ThemeColorBlind.YELLOW))
        addVerticalStrut()
        addHeader(Game.PATH_OF_EXILE_1.explicitName)
        addComponent(poe1UsingFoldersCheckbox)
        addComponent(poe1tradeTabsInsideFolder)
        addVerticalStrut()
        addHeader(Game.PATH_OF_EXILE_2.toString())
        addComponent(poe2UsingFoldersCheckbox)
        addComponent(poe2tradeTabsInsideFolder)
    }

    protected override fun addComponentListeners() {
        poe1UsingFoldersCheckbox.onToggle {
            poe1tradeTabsInsideFolder.isEnabled = it
            runSetupValidation()
        }
        poe2UsingFoldersCheckbox.onToggle {
            poe2tradeTabsInsideFolder.isEnabled = it
            runSetupValidation()
        }
    }

    override fun initializeComponents() {
        poe1UsingFoldersCheckbox.isSelected = SaveManager.settingsSaveFile.data.settingsPoe1.usingStashFolder
        poe1tradeTabsInsideFolder.isEnabled = SaveManager.settingsSaveFile.data.settingsPoe1.usingStashFolder
        poe1tradeTabsInsideFolder.isSelected = SaveManager.settingsSaveFile.data.settingsPoe1.tradesAreInsideFolders
        poe2UsingFoldersCheckbox.isSelected = SaveManager.settingsSaveFile.data.settingsPoe2.usingStashFolder
        poe2tradeTabsInsideFolder.isEnabled = SaveManager.settingsSaveFile.data.settingsPoe2.usingStashFolder
        poe2tradeTabsInsideFolder.isSelected = SaveManager.settingsSaveFile.data.settingsPoe2.tradesAreInsideFolders
    }

    override fun isSetupValid(): Boolean {
        return true
    }

    override fun applyCompletedSetup() {
        SaveManager.settingsSaveFile.data.settingsPoe1.usingStashFolder = poe1UsingFoldersCheckbox.isSelected
        SaveManager.settingsSaveFile.data.settingsPoe1.tradesAreInsideFolders = poe1tradeTabsInsideFolder.isSelected
        SaveManager.settingsSaveFile.data.settingsPoe2.usingStashFolder = poe2UsingFoldersCheckbox.isSelected
        SaveManager.settingsSaveFile.data.settingsPoe2.tradesAreInsideFolders = poe2tradeTabsInsideFolder.isSelected
        SaveManager.settingsSaveFile.data.hasInitUsingStashFolders = true
        FrameManager.stashHelperContainerPoe1.updateBounds()
        FrameManager.stashHelperContainerPoe2.updateBounds()
    }

}
